tests: net: socket: net_mgmt move to new ztest API

hunterhu3000 opened the pull request:
Move net_mgmt tests to use new ztest API
TEST=twister -T tests/net/socket/net_mgmt

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

nashif commented on tests/net/socket/net_mgmt/src/main.c:
```diff
@@ -447,17 +447,17 @@ static void test_net_mgmt_catch_events(void)
 }
 
-static void test_net_mgmt_catch_kernel(void)
+ZTEST(socket_net_mgmt_test_suite, test_net_mgmt_catch_kernel)
```
no need for _test_suite here.

hunterhu3000 replied:
Updated. Thanks to review.

hunterhu3000 committed:
tests: net: socket: net_mgmt move to new ztest API

Move net_mgmt tests to use new ztest API
TEST=twister -T tests/net/socket/net_mgmt

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

hunterhu3000 commented:
This test case cannot be run with CONFIG_ZTEST_SHUFFLE =y, as some of the test case should be run with specific order.
